### PR TITLE
Changes AddVocabDrift to have no-drift option

### PR DESCRIPTION
**Pull request by JamesWardAntony**

### CODE REVIEW

File: patgen/configvocab.go [hunk 1]
```diff
@@ -93,10 +93,11 @@ func AddVocabClone(mp Vocab, name string, copyFrom string) (*etensor.Float32, er
 // AddVocabRepeat adds a repeated pool to the vocabulary,
 // copying from given row in existing vocabulary item .
 func AddVocabRepeat(mp Vocab, name string, rows int, copyFrom string, copyRow int) (*etensor.Float32, error) {
-	cp, err := mp.ByNameTry(copyFrom)
+	origItem, err := mp.ByNameTry(copyFrom)
 	if err != nil {
 		return nil, err
 	}
+	cp := origItem.Clone()
```
Comment (rcoreilly) [inline patgen/configvocab.go]: yeah probably we don't need to clone it, because it is copied anyway -- why did you change then revert this?

File: patgen/configvocab.go [hunk 2]
```diff
@@ -131,12 +130,18 @@ func AddVocabDrift(mp Vocab, name string, rows int, pctDrift float32, copyFrom s
 	trow.CopyFrom(cprow)
 	nOn := NOnInTensor(cprow)
 	nDrift := int(math.Round(float64(nOn) * float64(pctDrift)))
-	nDrift = ints.MaxInt(1, nDrift) // ensure at least one
+	//nDrift = ints.MaxInt(1, nDrift) // ensure at least one //JWA, was uncommented
+	nDrift = ints.MaxInt(0, nDrift) // ensure at least one //JWA, new
```
Comment (rcoreilly) [inline patgen/configvocab.go:135]: Here's a version that deals with "aliasing" for fractional drift levels, including case where drift < 1:

```Go
rmdr := 0.0 // remainder carryover in drift
drift := float64(nOn) * float64(pctDrift+derr) // precise fractional amount of drift
for i := 1; i < rows; i++ {
...
curDrift := math.Round(float64(nOn) * (float64(pctDrift)+rmdr))) // integer amount
nDrift = int(curDrift)
if nDrift > 0 {
  FlipBigts(trow, nDrift, nDrift, 1, 0)
}
rmdr += drift - curDrift // accumulate remainder
}
```
same logic as: https://en.wikipedia.org/wiki/Bresenham%27s_line_algorithm

**Comment by rcoreilly** [comment]
ps. above is untested, just off-the-top-of-my-head so you should fix it up and test it, then resubmit working version!

File: patgen/configvocab.go (hunk 1 above)
Comment (JamesWardAntony) [reply patgen/configvocab.go]: I just had the old version (I think) and wanted to update to the most recent one. Apologies for confusion...

File: patgen/configvocab.go (hunk 2 above)
Comment (JamesWardAntony) [reply patgen/configvocab.go:135]: Is that a typo, 'FlipBigts'?
